### DISCUSSION TITLE
What does the Trace config point 'stepDefinitionSnippetStyle' actually do?

### DISCUSSION THREAD

**Discussion by timothy-plummer-ul** [reqnroll/Reqnroll#159] [Q&A]
I have used the Reqnroll extension for a while now (I'm not sure how long, but more than a few weeks), initially as a backward compatibility test for SpecFlow and recently with full Reqnroll 2.0.1.

When I was using Reqnroll with SpecFlow, it would generate the step definition with Regex, but now that I'm using Reqnroll, it generates step definitions with Cucumber Expressions. I would prefer to work with Regex as that is what the rest of my project has been written in. 
I thought that setting  `"trace": { "stepDefinitionSnippetStyle": "RegexAttribute"}`  in the `reqnroll.json` file would have made it so that when you define a step using the Reqnroll for VS 2022 extension, it will define steps using Regex, not Cucumber Expressions, but I was wrong. 

My question is twofold:
1. What does the  `"trace": { "stepDefinitionSnippetStyle": "RegexAttribute"}` do in the reqnroll.json file?
2. How can I ensure that the Reqnroll extension generates the step definitions in Regex format as it did with SpecFlow?

**Comment by gasparnagy** [comment]
This should work. So if not, than this must be a bug. Could you please double check with a simple project, like the [quickstart sample](https://github.com/reqnroll/Quickstart) and create an issue if it does not work?

**Reply by timothy-plummer-ul** [reply]
A coworker and I just now verified this; we are wondering which repo you would like the issue posted in. We will post the sample app in the issue.

**Reply by clrudolphi** [reply]
I can confirm the behavior using the Quickstart project. It's not just the stepDefinitionSnippetStyle attribute, it seems the entire process of deserializing the reqnroll.json is broken. No settings in the reqnroll.json file are getting applied to the configuration object. After reading in and deserializing the json file, the config.Reqnroll.IsReqnrollProject is false and the Traits are null.

I'll keep looking for root. Did we recently update our Newtonsoft dependency version? Perhaps there was a breaking change there...

**Reply by gasparnagy** [reply]
See https://github.com/reqnroll/Reqnroll.VisualStudio/issues/18
